master/internal/api_searches.go

```diff
@@ -0,0 +1,656 @@
+package internal
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/uptrace/bun"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/internal/experiment"
+	"github.com/determined-ai/determined/master/internal/grpcutil"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/set"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/rbacv1"
+)
+
+type searchCandidateResult struct {
+	Archived   bool
+	ID         int32
+	IsTerminal bool
+}
+
+func filterSearchQuery(getQ *bun.SelectQuery, filter *string) (*bun.SelectQuery, error) {
```

Comment:
Is there any difference between this function and the one in `api_runs.go`? If not maybe we can import it, or move it to a util file.

Reply:
The only difference is that since we're not querying the runs table to filter searches, we only look at `WHERE NOT e.archived`. Kind of a tricky thing to factor out but I'm open to thinking more on how we might consolidate this.